### ISSUE TITLE
Add a map scale bar control

### ISSUE THREAD

**Issue by barbeau** [googlemaps/android-maps-utils#701]
**Is your feature request related to a problem? Please describe.**
It would be nice to have the ability to easily add a map scale bar showing the current scale of the map in meters and feet.

The current Google Maps app behavior as of April 2020 is that the map scale bar is hidden by default. It only appears when you zoom in or out, and then stays on the screen for about 3 seconds after zooming stops before it fades to invisible:

![image](https://user-images.githubusercontent.com/928045/80747025-c29bdc80-8af0-11ea-8e60-e2f6adbf5cf4.png)

It looks map scale control is supported natively in the Maps JavaScript API:
https://developers.google.com/maps/documentation/javascript/controls

..but not Android.

 **Describe the solution you'd like**
Add a simple map scale bar control in the utility library that could be toggeled on or off with a few lines of code

 **Describe alternatives you've considered**
Do nothing - have developers use third-party solutions like https://github.com/pengrad/MapScaleView.

**Comment by barbeau** [comment]
Still valid

**Comment by barbeau** [comment]
@arriolac for a UI component of the library like this (map scale bar control), what are your thoughts on using [Jetpack Compose](https://developer.android.com/jetpack/compose) to implement?

**Comment by arriolac** [comment]
I think that's a great idea especially since there are [Interoperability APIs](https://developer.android.com/jetpack/compose/interop/interop-apis) available to go between Android Views and Compose. 

Do you know how large an app's binary would increase by should we use Compose?

**Comment by barbeau** [comment]
This page has some really good hard numbers for two of the demo apps:
https://developer.android.com/jetpack/compose/ergonomics

>* Tivi fully migrated to Compose, removing AppCompat and Material Design Components libraries.
>* Sunflower (compose_recyclerview branch) adds Compose only using it for items of a RecyclerView. All other dependencies are the same.
>
>...
>Tivi saw a 46% reduction in APK size, from 4.49 MB to 2.39 MB. It also saw a 17% reduction in method count after fully migrating to Compose. The XML lines of code decreased by 76%.
>
>Sunflower's APK size increased 575 KB when Compose was added to the project, from 2407 KB to 2982 KB. Since Compose is barely used in that Sunflower branch, the project doesn't benefit from Compose-only APK size benefits such as removing the AppCompat dependency from the project.

![image](https://user-images.githubusercontent.com/928045/133514983-2550a521-56cb-4498-918f-9772a45323ed.png)

So overhead would be for apps that aren't using Compose at all, and maybe around 575 KB? There wouldn't be any overhead for apps already using Compose (partially or completely).

**Comment by arriolac** [comment]
Good to know!

Are there any other views you think that can be written in Compose? Wondering if we should consider writing it in a separate compose-specific library.

**Comment by barbeau** [comment]
Yeah, that's a good question. 

There are only a few layouts in AMU related to info windows in clustering and KML/GeoJSON rendering, and they are very simple (and I think `amu_webview.xml` is maybe unused? Android Studio doesn't show any usages and if I delete it the project still builds):

![image](https://user-images.githubusercontent.com/928045/133681461-3e665daf-2bf4-4aa5-b472-562980b3230d.png)

I don't think these are worth converting to Compose for their own sakes.

I don't think 3D follow (https://github.com/googlemaps/android-maps-utils/issues/714) or arrows along polygons (https://github.com/googlemaps/android-maps-utils/issues/702) would require their own views/layouts.

So the map scale bar control might be the lone contender that requires it's own layouts/views worthy of Compose. It would probably depend on maps-ktx.

Maybe it's best in it's own maps-widgets (?) library so developers don't incur the overhead unless they're using it specifically?

**Comment by arriolac** [comment]
Yeah I think it would be best to put it in its own library for that reason. Development can start in this repo as a new module, and once it's close to a releasable state, I can help create a new repo (it has to go through a couple approvals internally).

**Comment by barbeau** [comment]
Sounds good!

**Comment by barbeau** [comment]
I'm working on this and have a version I'll share next week.

**Comment by barbeau** [comment]
A draft PR implementation of the map scale bar is here on the android-maps-ktx library:
https://github.com/googlemaps/android-maps-ktx/pull/175

**Comment by thomaskroth** [comment]
Would love to see this feature

**Comment by barbeau** [comment]
A PR to implement this as part of android-maps-compose library is now open at https://github.com/googlemaps/android-maps-compose/pull/157.
